src/screen-cast.c

```diff
@@ -1002,11 +1002,9 @@ replace_restore_data_by_token (ScreenCastSession *screen_cast_session,
     {
       if (g_strcmp0 (key, "restore_data") == 0)
         {
-          GVariant *child = g_variant_get_child_value (value, 0);
```

Comment:
I don't think this is correct. It is expected that portal implementations provide a variant of signature `v` here, with a single child containing the `(suv)` tuple. That's because we're iterating over an `a{sv}` vardict.

Reply:
`persist_mode` is also in the `a{sv}` and we are not calling g_variant_get_child_value there.

Reply:
Right

*sighs*

I don't know why I still try to code, this is clearly not my craft.